pygmt/tests/test_clib_read_data.py

```diff
@@ -132,6 +132,60 @@ def test_clib_read_data_grid_actual_image():
             )
 
 
+# Note: Simplify the tests for images after GMT_IMAGE.to_dataarray() is implemented.
+def test_clib_read_data_image():
+    """
+    Test the Session.read_data method for images.
+    """
+    with Session() as lib:
+        image = lib.read_data("@earth_day_01d_p", kind="image").contents
+        header = image.header.contents
+        assert header.n_rows == 180
+        assert header.n_columns == 360
+        assert header.n_bands == 3
+        assert header.wesn[:] == [-180.0, 180.0, -90.0, 90.0]
+        assert image.data
+
+
+def test_clib_read_data_image_two_steps():
+    """
+    Test the Session.read_data method for images in two steps, first reading the header
+    and then the data.
+    """
+    infile = "@earth_day_01d_p"
+    with Session() as lib:
+        # Read the header first
+        data_ptr = lib.read_data(infile, kind="image", mode="GMT_CONTAINER_ONLY")
+        grid = data_ptr.contents
+        header = grid.header.contents
+        assert header.n_rows == 180
+        assert header.n_columns == 360
+        assert header.wesn[:] == [-180.0, 180.0, -90.0, 90.0]
+        assert header.n_bands == 3  # Explicitly check n_bands
+        assert not grid.data  # The data is not read yet
+
+        # Read the data
+        lib.read_data(infile, kind="image", mode="GMT_DATA_ONLY", data=data_ptr)
+        assert grid.data
+
+
+def test_clib_read_data_image_actual_grid():
+    """
+    Test the Session.read_data method for image, but actually the file is a grid.
+    """
+    with Session() as lib:
+        data_ptr = lib.read_data(
+            "@earth_relief_01d_p", kind="image", mode="GMT_CONTAINER_ONLY"
+        )
```

Comment:
Getting this error at https://github.com/GenericMappingTools/pygmt/actions/runs/10017032324/job/27690729251?pr=3338#step:8:1556:

```python-traceback
ERROR 4: `/home/runner/.gmt/server/earth/earth_relief/earth_relief_01d_p.grd' not recognized as being in a supported file format. It could have been recognized by driver HDF5, but plugin gdal_HDF5.so is not available in your installation. You may install it with 'conda install -c conda-forge libgdal-hdf5'
Error: ession [ERROR]: Unable to open earth_relief_01d_p.grd.
Error: ession [ERROR]: ERROR reading image with gdalread.
pygmt-session (gmtapi_import_image): Could not read from file [earth_relief_01d_p.grd]
[Session pygmt-session (198)]: Error returned from GMT API: GMT_IMAGE_READ_ERROR (22)
[Session pygmt-session (198)]: Error returned from GMT API: GMT_IMAGE_READ_ERROR (22)
```

So do we need to add `libgdal-hdf5` to gmt-feedstock, similar to https://github.com/conda-forge/gmt-feedstock/pull/291?

Reply:
I think yes

Reply:
Working in https://github.com/conda-forge/gmt-feedstock/pull/293.

Reply:
This test still fails even if `libgdal-hdf5` is installed, see https://github.com/GenericMappingTools/pygmt/actions/runs/10018688146/job/27694518606?pr=3338.

The expected `header.wesn` is [-180, 180, -90, 90], but the actual results are platform-dependent. For Python 3.12 + Ubuntu, the result is `[-179.5, 179.5, -89.5, 89.5]` but for Python 3.10 + Ubuntu, the result is `[0.5, 359.5, 0.5, 179.5]`.
```
    def test_clib_read_data_image_actual_grid():
        """
        Test the Session.read_data method for image, but actually the file is a grid.
        """
        with Session() as lib:
            data_ptr = lib.read_data(
                "@earth_relief_01d_p", kind="image", mode="GMT_CONTAINER_ONLY"
            )
            image = data_ptr.contents
            header = image.header.contents
            assert header.n_rows == 180
            assert header.n_columns == 360
>           assert header.wesn[:] == [-179.5, 179.5, -89.5, 89.5]
E           assert [0.5, 359.5, 0.5, 179.5] == [-179.5, 179.5, -89.5, 89.5]
```
The test added in https://github.com/conda-forge/gmt-feedstock/pull/293 also fails (see https://github.com/conda-forge/gmt-feedstock/pull/293/checks?check_run_id=27696860313, https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=983672&view=logs&jobId=656edd35-690f-5c53-9ba3-09c10d0bea97&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=986b1512-c876-5f92-0d81-ba851554a0a3). 

Not interested in digging down the rabbit hole, so I prefer to removing this test.

Reply:
Hmm, or we can keep the test and mark it with xfail? Might have something to do with different GDAL versions or something. I almost think we should include GDAL in `pygmt.show_versions()` to more easily check things.

Reply:
I've added the test back but the macOS + Python 3.12 CI job keeps crashing (https://github.com/GenericMappingTools/pygmt/actions/runs/10024733960/job/27707337991?pr=3338).